core/port/availability.go

```diff
@@ -22,32 +22,14 @@ import (
 	"strconv"
 
 	log "github.com/cihub/seelog"
-	"github.com/pkg/errors"
 )
 
-func available(protocol string, port int) (bool, error) {
-	addr := ":" + strconv.Itoa(port)
-
-	switch protocol {
-	case "udp", "udp4", "udp6":
-		udpAddr, err := net.ResolveUDPAddr(protocol, addr)
-		if err != nil {
-			return false, errors.Wrap(err, "unable to resolve UDP address")
-		}
-		conn, err := net.ListenUDP(protocol, udpAddr)
-		if err != nil {
-			log.Infof("%s cannot listen on UDP port %v: %v", logPrefix, port, err)
-			return false, nil
-		}
-		defer conn.Close()
-	default:
-		listener, err := net.Listen(protocol, addr)
-		if err != nil {
-			log.Infof("%s cannot listen on TCP port %v: %v", logPrefix, port, err)
-			return false, nil
-		}
-		defer listener.Close()
+func available(port int) bool {
+	listener, err := net.Listen("tcp", ":"+strconv.Itoa(port))
+	if err != nil {
+		log.Infof("%s cannot listen on TCP port %v: %v", logPrefix, port, err)
```

Comment:
I would write here TCP, since for us it does not matter. We just need to know if port is free.
```suggestion
		log.Infof("%s port %v: is taken %v", logPrefix, port, err)
```

Reply:
Missed this, thanks.